File: backend/src/routes/books.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import { listarLivros, criarLivro, deletarLivro } from '../controllers/books.controller'

const router = Router()

router.use(requireAuth)

// GET  /api/books     → lista livros do usuário
// POST /api/books     → adiciona novo livro
router.get('/',  listarLivros)
router.post('/', criarLivro)

// DELETE /api/books/:id  → remove livro
router.delete('/:id', deletarLivro)

export default router

File: backend/internal/books/books_routes.go
package books

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de livros no grupo /api/books. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("", ctrl.ListarLivros)
	router.POST("", ctrl.CriarLivro)
	router.DELETE("/:id", ctrl.DeletarLivro)
}
